apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -293,7 +296,46 @@ impl MappedResponse {
                         }
                     };
                 }
-                ResponseKey::BatchEntity { .. } => {} // TODO $batch
+                ResponseKey::BatchEntity { keys, inputs, .. } => {
+                    if let serde_json_bytes::Value::Array(values) = value {
+                        let key_selection = keys
+                            .to_mapping()
+                            .map_err(|e| HandleResponseError::MergeError(e.to_string()))?;
+
+                        // Convert representations into keys for use in the map
+                        let key_values = inputs
+                            .batch
+                            .iter()
+                            .map(|v| {
+                                key_selection
+                                    .apply_to(&Value::Object(v.clone()))
+                                    .0
+                                    .unwrap_or(Value::Null)
+                            })
+                            .collect_vec();
+
+                        // Create a map of keys to entities
+                        let mut map = values
+                            .into_iter()
+                            .map(|v| {
+                                let key = key_selection.apply_to(&v).0.unwrap_or_default();
+                                (key, v)
+                            })
+                            .collect::<HashMap<_, _>>();
+
+                        // Make a list of entities that matches the representations list
+                        let entities = key_values
+                            .iter()
+                            .map(|key| map.remove(key).unwrap_or(Value::Null))
```

Comment:
```suggestion
                            .map(|key| map.remove(&key).unwrap_or(Value::Null))
```
If it's not a vec above, than this should be the correct modification